modules/tiles/src/tileset/traversers/tileset-traverser.js

```diff
@@ -145,6 +144,8 @@ export default class TilesetTraverser {
       !skipLevelOfDetail && tile.refine === TILE_REFINEMENT.REPLACE && tile.hasRenderContent;
 
     let hasVisibleChild = false;
+    let refines = true;
```

Comment:
Should we make refinement calculation into a separate helper function? And ideally add a unit test case?

Reply:
Hi, about a helper function, at least for me it helps to try and mirror the [CesiumJS code](https://github.com/CesiumGS/cesium/blob/master/Source/Scene/Cesium3DTilesetTraversal.js#L441-L529) as much as possible so I can somewhat keep track of the differences.
Unit testing would definitely be good, but I will not have the time to add that soon.